appinventor/buildserver/src/com/google/appinventor/buildserver/Project.java

```diff
@@ -145,11 +146,8 @@ public Project(File file) {
 
       // Load project file
       properties = new Properties();
-      FileInputStream in = new FileInputStream(file);
-      try {
+      try (InputStreamReader in = new InputStreamReader(new FileInputStream(file))) {
         properties.load(in);
-      } finally {
-        in.close();
```

Comment:
@ewpatton You do not have to close the InputStreamReader (or the interior FIleInputStream)?

Reply:
This update makes use of a Java 1.7 feature called try-with-resources. When the try block is exited, the resources created in the parens after the try are automatically closed (even in the case of an exception). Effectively, it's just syntactic sugar that behaves the same as the old w.r.t. the finally block. The reader's close method will call the inner stream's close method.